src/cocoa/toga_cocoa/dialogs.py

```diff
@@ -132,4 +132,33 @@ def open_file(window, title, file_types, multiselect):
         filename_or_filenames = (paths if multiselect else
                                  panel.URL.path)
         return filename_or_filenames
-    return None
+
+
+def select_folder(window, title, multiselect):
+    """Cocoa select folder dialog implementation.
+
+    Args:
+        window: Window dialog belongs to.
+        title: Title of the dialog.
+        multiselect: Flag to allow multiple folder selection.
+    Returns:
+        (list) Paths of folders
+    """
+    dialog = NSOpenPanel.alloc().init()
```

Comment:
At column 14: (F405) 'NSOpenPanel' may be undefined, or defined from star imports: .libs